sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs

```diff
@@ -1725,8 +1753,15 @@ private void SetNameFieldsIfNull()
             long length = response.IsUnavailable() ? 0 : response.Headers.ContentLength ?? 0;
             ClientConfiguration.Pipeline.LogTrace($"Response: {response.GetRawResponse().Status}, ContentLength: {length}");
 
+            BlobDownloadStreamingResult result = response.ToBlobDownloadStreamingResult();
+            if (response.GetRawResponse().Headers.TryGetValue(Constants.StructuredMessage.CrcStructuredMessageHeader, out string _) &&
```

Comment:
Do we want to fail if we asked for structured message but didn't get one back? Again, not sure here, just a thought.

Reply:
Probably. Need to think on that more. Good catch.